service/actions/general.js

```diff
@@ -2,17 +2,28 @@
 // See License.txt for license information.
 
 import Client from 'service/client';
-import {bindClientFunc} from './helpers.js';
+import {bindClientFunc, FormattedError} from './helpers.js';
 import {GeneralTypes} from 'service/constants';
 import {getMyChannelMembers} from './channels';
 
 export function getPing() {
-    return bindClientFunc(
-        Client.getPing,
-        GeneralTypes.PING_REQUEST,
-        GeneralTypes.PING_SUCCESS,
-        GeneralTypes.PING_FAILURE
-    );
+    return async (dispatch, getState) => {
+        dispatch({type: GeneralTypes.PING_REQUEST}, getState);
+
+        let data;
+        try {
+            data = await Client.getPing();
+            if (!data.version) {
+                // successful ping but not the right return data
+                throw new FormattedError('error.network_request_failed', 'Network request failed');
```

Comment:
Actually I'm thinking that this message should be changed to something like `Could not connect to the server, make sure you have an internet connection and the correct URL` but it might be too long, @lfbrock thoughts?